src/Microsoft.AspNetCore.SignalR/HubConnectionList.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.SignalR
+{
+    public class HubConnectionList : IReadOnlyCollection<HubConnectionContext>
```

Comment:
Well, if you're going to strawman the name, then it sounds ridiculous, but I'm not so sure it's a bad idea. Something like `IConnectionContext` with a `ConnectionId` property. Or a utility base class that exposes a single `protected abstract` method like `GetConnectionId` (kinda like `KeyedCollection` does)